examples/multi-agent-debate/main.py

```diff
@@ -0,0 +1,391 @@
+import typer
+import json
+import logging
+from rich.prompt import Prompt, Confirm
+from typing import List, Tuple, Optional, Literal, Any
+import langroid as lr
+from langroid.language_models import OpenAIGPTConfig
+from langroid.agent.chat_agent import ChatAgent, ChatAgentConfig
+from langroid.agent.task import Task
+from langroid.agent.tools.orchestration import DoneTool
+from langroid.agent.tools.google_search_tool import GoogleSearchTool
+from langroid.utils.logging import setup_logger
+from langroid.utils.configuration import set_global
+from config import get_base_llm_config, get_global_settings
+from models import SystemMessages, Message
+
+DEFAULT_SYSTEM_MESSAGE_ADDITION = f"""
```

Comment:
This will _not_ work no matter how much we plead 😀 - this is because LLMs have no ability to differentiate real from fake URLs. Also you are asking it to use sources from a web search, but you are not enabling the agent with a web-search tool (you're only enabling it for the validation agent at the end). A better way to set this up is to enable both the debate agents to use a web-search tool. The `task.run()` loop handles a web-search tool call, returns results to the LLM. Your sys msgs will have to instruct the LLM to support its args with references from a web search, indicating which tool to use (see comment below). Also when the sub-task agent uses a tool, `single_round=True` won't work (since it will just get llm response and return, without handling the tool), so you have to instruct it to use the DoneTool when done. You can search for `DoneTool` and `MetaphorSearchTool` under `examples` to see how those are used. All in all, the addition of a tool somewhat complicates the setup, so it is up to you if you want to get into it. (tip - install the `ripgrep` tool for fast command-line search using `rg MetaphorSearchTool -t py`)

Reply:
This makes sense. Got the key for MetaphorSearchTool. Will implement in the future V2 version of debate system.  For right now I removed the reference to Web Search.